From 213490099b7d24664ac635f25c9cf224d6a3e702 Mon Sep 17 00:00:00 2001
From: Benjamin Otte <otte@redhat.com>
Date: Thu, 26 May 2022 04:36:14 +0200
Subject: [PATCH] x11: Always update shadow size

Not updating shadow size unconditionally would lead to shadow size not
being set on map, which would lead mutter to think that we are a Window
without extents and then become confused when we suddenly set some.

Make sure that doesn't happen by always having shadows set on map, just
like GTK3.

Fixes #4136
---
 gdk/x11/gdksurface-x11.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gdk/x11/gdksurface-x11.c b/gdk/x11/gdksurface-x11.c
index 8567d83eb8..fb3b0bd7df 100644
--- a/gdk/x11/gdksurface-x11.c
+++ b/gdk/x11/gdksurface-x11.c
@@ -280,7 +280,7 @@ compute_toplevel_size (GdkSurface *surface,
   gdk_toplevel_size_init (&size, bounds_width, bounds_height);
   gdk_toplevel_notify_compute_size (GDK_TOPLEVEL (surface), &size);
 
-  if (size.shadow.is_valid && update_geometry)
+  if (size.shadow.is_valid)
     {
       update_shadow_size (surface,
                           size.shadow.left,
-- 
GitLab

